metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/AbstractAllAtOnceStrategy.java

```diff
@@ -87,6 +88,26 @@ protected AbstractAllAtOnceStrategy(
 	 * @param hostname
 	 */
 	private void process(final Connector currentConnector, final String hostname) {
+		if (!validateConnectorDetectionCriteria(currentConnector, hostname)) {
+			log.error(
+				"Hostname {} - The connector {} no longer matches the host. Stopping the connector's collect job.",
+				hostname,
+				currentConnector.getCompiledFilename()
+			);
+			return;
+		}
+
+		// Run PreSourceStrategy that executes pre sources
+
+		final PreSourceStrategy preSourceStrategy = PreSourceStrategy
```

Comment:
preSourceStrategy needs to be instantiated with `currentConnector`. Then it should  process the `currentConnector`'s pre sources only.